cli/src/complete.rs

```diff
@@ -44,6 +46,112 @@ pub fn local_bookmarks() -> Vec<CompletionCandidate> {
     })
 }
 
+pub fn tracked_bookmarks() -> Vec<CompletionCandidate> {
+    with_jj(|mut jj, _| {
+        let output = jj
+            .arg("bookmark")
+            .arg("list")
+            .arg("--tracked")
+            .arg("--template")
+            .arg(r#"if(remote, name ++ "@" ++ remote ++ "\n")"#)
+            .output()
+            .map_err(user_error)?;
+
+        Ok(String::from_utf8_lossy(&output.stdout)
+            .lines()
+            .map(CompletionCandidate::new)
+            .collect())
+    })
+}
+
+pub fn untracked_bookmarks() -> Vec<CompletionCandidate> {
+    with_jj(|mut jj, config| {
+        let output = jj
+            .arg("bookmark")
+            .arg("list")
+            .arg("--all-remotes")
+            .arg("--template")
+            .arg(r#"if(remote && !tracked, name ++ "@" ++ remote ++ "\n")"#)
+            .output()
+            .map_err(user_error)?;
+
+        Ok(String::from_utf8_lossy(&output.stdout)
+            .lines()
+            .filter(|bookmark| !bookmark.ends_with("@git"))
+            .map(|bookmark| {
+                let mut display_order = 0;
+                if let Ok(prefix) = config.get::<String>("git.push-bookmark-prefix") {
+                    if !bookmark.starts_with(&prefix) {
+                        // other people's bookmarks are less interesting
+                        display_order += 1;
+                    }
+                }
+                CompletionCandidate::new(bookmark).display_order(Some(display_order))
+            })
+            .collect())
+    })
+}
+
+pub fn bookmarks() -> Vec<CompletionCandidate> {
+    with_jj(|mut jj, config| {
+        let output = jj
+            .arg("bookmark")
+            .arg("list")
+            .arg("--all-remotes")
+            .arg("--template")
+            .arg(r#"separate("@", name, remote) ++ "\n""#)
+            .output()
+            .map_err(user_error)?;
+        let stdout = String::from_utf8_lossy(&output.stdout);
+
+        let prefix = config.get::<String>("git.push-bookmark-prefix").ok();
+        let mut bookmarks = HashMap::new();
+
+        // To determine the importance of each bookmark, we figure out if it is
+        // tracked locally and if its push prefix matches our configured one.
+        for line in stdout.lines() {
+            let (bookmark, _) = line.split_once('@').unwrap_or((line, ""));
```

Comment:
nit: better to use the split result instead of testing `line.contains('@')` again.